boards/riscv32/hifive1/hifive1.dts

```diff
@@ -52,6 +53,13 @@
 &spi0 {
 	status = "ok";
 	clock-frequency = <16000000>;
+
+	#address-cells = <1>;
+	#size-cells = <1>;
+	flash0: flash@20400000 {
+		compatible = "soc-nv-flash";
+		reg = <0x20400000 0xc00000>;
+	};
```

Comment:
@galak please confirm, but I think this should be (assuming spi chip select 0):
```
	reg = <0x10014000 0x1000>, <0x20400000 0xc00000>;
	flash0: flash@0 {
		compatible = "soc-nv-flash";
		reg = <0>;
	};
```

Reply:
This still needs to be fixed

Reply:
I fixed this and the analogous dts for riscv32 QEMU platform.

Reply:
@galak which is the preferred syntax?
```reg = <0x10014000 0x1000>, <0x20400000 0xc00000>;```
or
```reg = <0x10014000 0x1000 0x20400000 0xc00000>;```

Reply:
@MaureenHelm 

I used the same notation as in the original [spi0](https://github.com/zephyrproject-rtos/zephyr/blob/master/dts/riscv32/riscv32-fe310.dtsi#L181) node just be consistent.